ivadomed/loader/utils.py

```diff
@@ -654,18 +661,20 @@ def create_bids_dataframe(self):
             # Drop rows with json, tsv and LICENSE files in case no extensions are provided in config file for filtering
             df_next = df_next[~df_next['filename'].str.endswith(tuple(['.json', '.tsv', 'LICENSE']))]
 
-            # Update dataframe with subject files of chosen contrasts and extensions,
-            # and with derivative files of chosen target_suffix from loader parameters
+            # Update dataframe with subject files of chosen contrasts
+            # and with derivative files of chosen target_suffix
             df_next = df_next[(~df_next['path'].str.contains('derivatives')
-                               & df_next['suffix'].str.contains('|'.join(self.contrast_lst))
-                               & df_next['extension'].str.contains('|'.join(self.extensions)))
+                               & df_next['suffix'].str.contains('|'.join(self.contrast_lst)))
                                | (df_next['path'].str.contains('derivatives')
                                & df_next['filename'].str.contains('|'.join(self.target_suffix)))]
 
+            # Update dataframe with files of chosen extensions if present
+            if self.extensions:
+                df_next = df_next[df_next['filename'].str.endswith(tuple(self.extensions))]
```

Comment:
Nitpick: We can still use `df_next['extension']` here, right?

Reply:
No we cannot use `['extension']` here unfortunately. This is because of the parsing from `pybids`.

When the files are "raw data" in a `sub-` folder, `pybids` parses all the entities + extensions of the filenames.
But when the files are in the `derivatives` folder, they are indexed but the filenames are not parsed and the `['extension']` cells are empty.
Example [here](https://github.com/ivadomed/data-testing/blob/master/microscopy_png/df_ref.csv): `seg-*` files don't have metadata in the dataframe.

I changed it to have the filtering by extensions on both on `sub-` and `derivatives`.

Reply:
Thanks for the clarification! Sorry about that, that was a naive suggestion my part. :sweat_smile:

Reply:
I removed the `if self.extensions` condition because now it is always true with `.nii` and `.nii.gz` by default.